### PR TITLE
efficient implementation for multiple merger

### PR DESCRIPTION
**Pull request by shajoezhu**
not huge amount of code reduction, but the logic is much clearer I think

### CODE REVIEW

File: lib/msprime.c [hunk 1]
```diff
@@ -3778,39 +3777,18 @@ msp_dirac_common_ancestor_event(msp_t *self, population_id_t pop_id)
...
-                avl_init_node(q_node, u);
-                /* Now assign this ancestor to a uniformly chosen pot */
-                j = (uint32_t) gsl_rng_uniform_int(self->rng, num_pots);
-                q_node = avl_insert_node(&Q[j], q_node);
-                assert(q_node != NULL);
-            }
-            node = next;
-        }
+
+        ret = msp_multi_merger_common_ancestor_event(self,
```
Comment (jeromekelleher) [inline lib/msprime.c]: Need to check ret here

Comment (shajoezhu) [reply lib/msprime.c]: oops... missed this one. fixing now

Comment (shajoezhu) [reply lib/msprime.c]: sorry

**Comment by jeromekelleher** [comment]
Very nice, thanks @shajoezhu !

**Comment by shajoezhu** [comment]
brilliant! Thanks @jeromekelleher !

**Pull request merged**